### PR TITLE
ROCm3.8 hipcub, hipsparse, rocprim, rocsparse & rocthrust

### PR DESCRIPTION
**Pull request by gkathirv**

### CODE REVIEW

**Comment by gkathirv** [comment]
@adamjstewart @srekolam @arjun-raj-kuppala

File: var/spack/repos/builtin/packages/hipcub/package.py [hunk 1]
```diff
@@ -14,14 +14,15 @@ class Hipcub(CMakePackage):
...
+    version('3.8.0', sha256='11d7d97268aeb953c34a80125c4577e27cb57cb6095606533105cecf2bd2ec9c')
     version('3.7.0', sha256='a2438632ea1606e83a8c0e1a8777aa5fdca66d77d90862642eb0ec2314b4978d')
     version('3.5.0', sha256='1eb2cb5f6e90ed1b7a9ac6dd86f09ec2ea27bceb5a92eeffa9c2123950c53b9d')
 
     variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
 
     depends_on('cmake@3:', type='build')
-    depends_on('numactl', when='@3.7.0')
-    for ver in ['3.5.0', '3.7.0']:
+    depends_on('numactl', type='link')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hipcub/package.py]: Is this needed for 3.5.0?

File: var/spack/repos/builtin/packages/rocm-clang-ocl/package.py [hunk 2]
```diff
@@ -10,19 +10,19 @@ class RocmClangOcl(CMakePackage):
...
 
     depends_on('cmake@3:', type='build')
 
-    for ver in ['3.5.0', '3.7.0']:
+    for ver in ['3.5.0', '3.7.0', '3.8.0']:
         depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
         depends_on('llvm-amdgpu@' + ver, type='build', when='@' + ver)
-
-    depends_on('rocm-device-libs@3.7.0', type='build', when='@3.7.0')
+        depends_on('rocm-device-libs@' + ver, type='build', when='@' + ver)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rocm-clang-ocl/package.py:29]: Is this needed for 3.5.0?

File: var/spack/repos/builtin/packages/rocprim/package.py [hunk 3]
```diff
@@ -10,18 +10,19 @@ class Rocprim(CMakePackage):
...
+    version('3.8.0', sha256='4d37320d174eaada99dd796d81fa97d5dcc65a6dff8e8ff1c21e8e68acb4ea74')
     version('3.7.0', sha256='225209a0cbd003c241821c8a9192cec5c07c7f1a6ab7da296305fc69f5f6d365')
     version('3.5.0', sha256='29302dbeb27ae88632aa1be43a721f03e7e597c329602f9ca9c9c530c1def40d')
 
     variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
 
     depends_on('cmake@3:', type='build')
-    depends_on('numactl', when='@3.7.0')
-    for ver in ['3.5.0', '3.7.0']:
+    depends_on('numactl', type='link')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rocprim/package.py]: Is this needed for 3.5.0?

**Review by adamjstewart — COMMENTED** [review]
@haampie may want to review this as well

File: var/spack/repos/builtin/packages/rocm-clang-ocl/package.py (hunk 2 above)
Comment (srekolam) [reply var/spack/repos/builtin/packages/rocm-clang-ocl/package.py:29]: no...it is not required for 3.5.0

File: var/spack/repos/builtin/packages/rocprim/package.py (hunk 3 above)
Comment (srekolam) [reply var/spack/repos/builtin/packages/rocprim/package.py]: numactl is not required for 3.5.0

File: var/spack/repos/builtin/packages/hipcub/package.py [hunk 4]
```diff
@@ -14,19 +14,21 @@ class Hipcub(CMakePackage):
...
-    depends_on('numactl', when='@3.7.0')
-    for ver in ['3.5.0', '3.7.0']:
+    for ver in ['3.5.0', '3.7.0', '3.8.0']:
         depends_on('hip@' + ver, type='build', when='@' + ver)
         depends_on('rocm-device-libs@' + ver, type='build', when='@' + ver)
         depends_on('comgr@' + ver, type='build', when='@' + ver)
         depends_on('hsa-rocr-dev@' + ver, type='build', when='@' + ver)
         depends_on('rocprim@' + ver, type='build', when='@' + ver)
+    for ver in ['3.7.0', '3.8.0']:
+        depends_on('numactl', type='link')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hipcub/package.py]:
```suggestion
    depends_on('numactl', type='link', when='@3.7.0:')
```

File: var/spack/repos/builtin/packages/hipsparse/package.py [hunk 5]
```diff
@@ -12,23 +12,25 @@ class Hipsparse(CMakePackage):
...
 
-    for ver in ['3.5.0', '3.7.0']:
+    for ver in ['3.5.0', '3.7.0', '3.8.0']:
         depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
         depends_on('rocm-device-libs@' + ver, type='build', when='@' + ver)
         depends_on('rocsparse@' + ver, type='build', when='@' + ver)
         depends_on('hip@' + ver, when='@' + ver)
         depends_on('comgr@' + ver, type='build', when='@' + ver)
         depends_on('hsa-rocr-dev@' + ver, type='link', when='@' + ver)
+        depends_on('rocprim@' + ver, type='link', when='@' + ver)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hipsparse/package.py:34]: Is this needed for 3.5.0 and 3.7.0?

File: var/spack/repos/builtin/packages/rocm-clang-ocl/package.py (hunk 2 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/rocm-clang-ocl/package.py:29]: In that case you should make another for-loop for it.

File: var/spack/repos/builtin/packages/rocprim/package.py [hunk 6]
```diff
@@ -10,22 +10,24 @@ class Rocprim(CMakePackage):
...
     depends_on('cmake@3:', type='build')
-    depends_on('numactl', when='@3.7.0')
-    for ver in ['3.5.0', '3.7.0']:
+    for ver in ['3.5.0', '3.7.0', '3.8.0']:
         depends_on('hip@' + ver, type='build', when='@' + ver)
         depends_on('rocm-device-libs@' + ver, type='build', when='@' + ver)
         depends_on('comgr@' + ver, type='build', when='@' + ver)
         depends_on('hsa-rocr-dev@' + ver, type='build', when='@' + ver)
+    for ver in ['3.7.0', '3.8.0']:
+        depends_on('numactl', type='link')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rocprim/package.py]:
```suggestion
    depends_on('numactl', type='link', when='@3.7.0:')
```

File: var/spack/repos/builtin/packages/hipsparse/package.py (hunk 5 above)
Comment (gkathirv) [reply var/spack/repos/builtin/packages/hipsparse/package.py:34]: no, only for 3.8.0

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/hipsparse/package.py:34]: Then make another for-loop

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
To fix the style and docs test, you'll need to rebase on develop now that #19018 has been merged. A new version of vermin was released that was more picky about Python 2/3 compatibility.

**Comment by gkathirv** [comment]
@adamjstewart Please merge this PR.

**Pull request merged**